src/Spout/Reader/CSV/RowIterator.php

```diff
@@ -148,23 +144,36 @@ public function next()
 
     /**
      * Returns the next line, converted if necessary to UTF-8.
-     * Neither fgets nor fgetcsv don't work with non UTF-8 data... so we need to do some things manually.
+     * As fgetcsv() does not manage correctly encoding for non UTF-8 data,
+     * we remove manually whitespace with ltrim or rtrim (depending on the order of the bytes)
      *
      * @return string|false The next line for the current file pointer, encoded in UTF-8 or FALSE if nothing to read
      * @throws \Box\Spout\Common\Exception\EncodingConversionException If unable to convert data to UTF-8
      */
     protected function getNextUTF8EncodedLine()
     {
-        // Read until the EOL delimiter or EOF is reached. The delimiter's encoding needs to match the CSV's encoding.
-        $encodedEOLDelimiter = $this->getEncodedEOLDelimiter();
-        $encodedLineData = $this->globalFunctionsHelper->stream_get_line($this->filePointer, self::MAX_READ_BYTES_PER_LINE, $encodedEOLDelimiter);
+        $encodedLineData = fgetcsv($this->filePointer, 0, $this->fieldDelimiter, $this->fieldEnclosure);
```

Comment:
don't use 0. Instead use self::MAX_READ_BYTES_PER_LINE. 0 defaults to 4096 bytes which may not be enough for some cells containing a very long string